### PR TITLE
Separate Event

### LINKED ISSUE #1426: Small feature request: rearrange tags in summary table
- Move none closer to wild tag since it is still a tag, somewhat.
- Event to the right since it is not really an on-going tag at all.
- Small thin line between separating None and Event from the rest of tags. Good to know how many tags you actually have.

image (10)

Tagging @chosta

### PR DESCRIPTION
**Pull request by chosta**
closes #1426

> Move none closer to wild tag since it is still a tag, somewhat.

@vsrisuknimit I'm not sure about moving `none` closer to `wild` just for the sake of it. It will complicate things in code if I do it anyway. But the separation would be helpful for those tag counting events and cards for sure!

the-best

### CODE REVIEW

File: src/components/Tag.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import Vue from 'vue';
+import {Tags} from '../cards/Tags';
 
 export const Tag = Vue.component('tag', {
   props: {
     tag: {
-      type: String,
+      type: String as () => Tags,
```
Comment (bafolts) [inline src/components/Tag.ts:7]: Sweet.

File: src/components/TagCount.ts [hunk 2]
```diff
@@ -29,6 +30,12 @@ export const TagCount = Vue.component('tag-count', {
       if (this.count === 0) {
         classes.push('tag-no-show');
       }
+      // event tag is added to the tail because it is not counted for actual tag for some cards and turmoil events (e.g. Interplanetary Trade)
+      if (this.tag === Tags.EVENT) {
+        classes.push('tag-event-separate');
+      }
+      console.log(this.type);
```
Comment (bafolts) [inline src/components/TagCount.ts:37]: This was probably added as debugging. Please remove before merging.

**Review by bafolts — APPROVED** [review]
Using `String as () => Tags` is great change! Spotted one `console.log` I would remove before merging.

**Comment by vsrisuknimit** [comment]
@chosta Looks great!

**Comment by kevinb9n** [comment]
@chosta Can we please please move the so-called "wild tag" to the other side of the line too? People need help understanding that it is NOT itself a tag. I was even talking to an experienced player who accidentally thought he had gotten to 9 tags for purposes of that turmoil event and really torpedoed his game.

File: src/cards/Tags.ts [hunk 3]
```diff
@@ -11,6 +11,6 @@ export enum Tags {
     ANIMAL = 'animal',
     CITY = 'city',
     WILDCARD = 'wild',
-    EVENT = 'event',
     MOON = 'moon',
+    EVENT = 'event', // please, always keep event at the tail
```
Comment (kberg) [inline src/cards/Tags.ts:15]: Why does event need to be at the end, specifically?

Comment (chosta) [reply src/cards/Tags.ts:15]: Because I'm lazy. I'll do it right in the next PR.

**Comment by chosta** [comment]
@kevinb9n @vsrisuknimit can we decide what needs to be before the separator?
I thought wild tag was counted towards Interplanetary Trade, if not maybe we should move it?

File: src/cards/Tags.ts (hunk 3 above)
Comment (kberg) [reply src/cards/Tags.ts:15]: The best response and worst response. (A good response.) OK, thanks.

File: src/components/TagCount.ts [hunk 4]
```diff
@@ -29,6 +30,11 @@ export const TagCount = Vue.component('tag-count', {
       if (this.count === 0) {
         classes.push('tag-no-show');
       }
+      // event tag is added to the tail because it is not counted for actual tag for some cards and turmoil events (e.g. Interplanetary Trade)
+      if (this.tag === Tags.EVENT) {
+        classes.push('tag-event-separate');
```
Comment (kberg) [inline src/components/TagCount.ts:35]: nit: consider a dedicated div as a separator.

**Comment by kevinb9n** [comment]
> @kevinb9n @vsrisuknimit can we decide what needs to be before the separator?
> I thought wild tag was counted towards Interplanetary Trade, if not maybe we should move it?

If you have all 10 tag types (Venus: 11), then a "wild tag" will *not* cause Interplanetary Trade to give you 11 mc production (Venus: 12). You can only make it become some real tag type, but you already have all of them.

**Pull request merged**